From bff30c14d6faaa3c0fa493718d072c4223e86a56 Mon Sep 17 00:00:00 2001
From: Thor Thayer <thor.thayer@linux.intel.com>
Date: Fri, 2 Feb 2018 09:24:43 -0600
Subject: [PATCH 027/151] FogBugz #532583-2: hwmon: Fix buffer calls for SDL
 compliance.

commit  b5f7278f6101fcbaa884bbd522c12003ba1edbaa from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

The buffer calls need to be more secure so pass the buffer size
in the sprintf functions. Limit buffer size to 5.

Signed-off-by: Thor Thayer <thor.thayer@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 drivers/hwmon/altera-a10sr-hwmon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwmon/altera-a10sr-hwmon.c b/drivers/hwmon/altera-a10sr-hwmon.c
index dac39a75fce2..584cc48a9b1b 100644
--- a/drivers/hwmon/altera-a10sr-hwmon.c
+++ b/drivers/hwmon/altera-a10sr-hwmon.c
@@ -126,7 +126,7 @@ static ssize_t altr_a10sr_read_status(struct device *dev,
 	else
 		val = !!(val & mask);
 
-	return sprintf(buf, "%d\n", val);
+	return scnprintf(buf, 5, "%d\n", val);
 }
 
 static ssize_t set_enable(struct device *dev,
-- 
2.26.1

